src/agent/multitype_queue/include/multitype_queue.hpp

```diff
@@ -34,21 +32,62 @@ class MultiTypeQueue : public IMultiTypeQueue
         {MessageType::STATEFUL, "STATEFUL"},
         {MessageType::COMMAND, "COMMAND"},
     };
-    const size_t m_maxItems;
-    const std::chrono::seconds m_timeout;
+
+    /// @brief maximun quantity of message to stored on the queue
+    size_t m_maxItems;
+
+    /// @brief timeout in milliseconds for refreshing the queue status
+    const std::chrono::milliseconds m_timeout;
+
+    /// @brief class for persistence implementation
     std::unique_ptr<Persistence> m_persistenceDest;
+
+    /// @brief mutex for protecting the queue access
     std::mutex m_mtx;
+
+    /// @brief condition variable related to the mutex
     std::condition_variable m_cv;
 
+    /// @brief Time between batch requests
+    std::time_t m_batchInterval = config::agent::DEFAULT_BATCH_INTERVAL;
+
 public:
     /**
      * @brief Constructor.
-     *
-     * @param dbFolderPath The path to the database folder.
-     * @param size The maximum number of items in the queue.
-     * @param timeout The timeout period in seconds.
-     */
-    MultiTypeQueue(const std::string& dbFolderPath, size_t size = DEFAULT_MAX, int timeout = DEFAULT_TIMEOUT_S);
+     * TODO:
+     * @param
+     */
+    template<typename ConfigGetter>
+    MultiTypeQueue(const ConfigGetter& getConfigValue)
+        : m_timeout(config::agent::QUEUE_STATUS_REFRESH_TIMER)
+    {
+        auto dbFolderPath =
+            getConfigValue.template operator()<std::string>("agent", "path.data").value_or(config::DEFAULT_DATA_PATH);
+
+        m_batchInterval = getConfigValue.template operator()<std::time_t>("events", "batch_interval")
+                              .value_or(config::agent::DEFAULT_BATCH_INTERVAL);
+
+        if (m_batchInterval < 1'000 || m_batchInterval > (1'000 * 60 * 60))
+        {
+            LogWarn("batch_interval must be between 1s and 1h. Using default value.");
+            m_batchInterval = config::agent::DEFAULT_BATCH_INTERVAL;
+        }
+
+        m_maxItems = getConfigValue.template operator()<size_t>("agent", "queue_size")
```

Comment:
include maximum and minimum validation for the number of events the queue can store,
we could use 1,000 as minimum and 100,000,000 as maximum.